compair/static/modules/image-viewer/image-viewer-directive.js

```diff
@@ -0,0 +1,273 @@
+(function() {
+
+var module = angular.module('ubc.ctlt.compair.image.viewer', []);
+
+// Directive for viewing images with rotation
+// Simplified version of https://github.com/fcrohas/angular-canvas-viewer
+module.directive('imageViewer', ['$window', '$q',
+    function($window, $q) {
+        var template =
+            '<div class="text-center image-viewer">' +
+                '<div class="btn btn-sm btn-info" ng-click="rotate(-1)"><span class="fa fa-rotate-left"></span></div>'+
+                '<div class="btn btn-sm btn-info" ng-click="rotate(1)"><span class="fa fa-rotate-right"></span></div>'+
+                '<br/>' +
+                '<i class="fa fa-spin fa-spinner" ng-show="!canvasLoaded"></i>' +
```

Comment:
Found the same

Reply:
I added the spinner because when testing it with a large image (~10MB), I was looking at a blank white space for two seconds after selecting the file.

But apparently `ng-show` isn't hiding the spinner quick enough.  So if the image is small, you will see both the spinner and image displayed at the same time.  I will wrap the spinner in `ng-if` [to avoid that](https://stackoverflow.com/a/33359816).